ironfish/src/network/peerNetwork.ts

```diff
@@ -600,30 +600,46 @@ export class PeerNetwork {
     })
   }
 
-  async getBlockHashes(peer: Peer, start: number, limit: number): Promise<Buffer[]> {
+  async getBlockHashes(peer: Peer, start: number, limit: number): Promise<[Buffer[], number]> {
+    const begin = BenchUtils.start()
+
     const message = new GetBlockHashesRequest(start, limit)
     const response = await this.requestFrom(peer, message)
 
+    if (response instanceof Error) {
+      throw response
+    }
+
     if (!(response.message instanceof GetBlockHashesResponse)) {
       // TODO jspafford: disconnect peer, or handle it more properly
       throw new Error(
         `Invalid GetBlockHashesResponse: ${displayNetworkMessageType(message.type)}`,
       )
     }
 
-    return response.message.hashes
+    return [response.message.hashes, BenchUtils.end(begin)]
   }
 
-  async getBlocks(peer: Peer, start: Buffer, limit: number): Promise<SerializedBlock[]> {
+  async getBlocks(
+    peer: Peer,
+    start: Buffer,
+    limit: number,
+  ): Promise<[SerializedBlock[], number] | Error> {
```

Comment:
I think the `[SerializedBlock[], number]` would be a little clearer if written as an object instead of an array.

Reply:
yea agreed, changed